src/metadata.ts

```diff
@@ -0,0 +1,220 @@
+const btoa = require('btoa');
+import { SERVER_URL } from './config';
+
+export enum Version {
+  v1,
+  v1w,
+  v2,
+}
+
+export interface MetadataInit {
+  name: string;
+  description?: string;
+  created_date: number;
+  registered_date?: Date | null;
+  expiration_date?: Date | null;
+  version: Version;
+}
+
+export interface Metadata {
+  name: string;
+  description?: string;
+  attributes: object[];
+  name_length?: number;
+  short_name?: string | null;
+  length?: number;
+  image_url?: string;
+  url?: string;
+  version: Version;
+}
+
+export class Metadata {
+  static MAX_CHAR = 30;
+  constructor({ name, description, created_date, version }: MetadataInit) {
+    this.name = name;
+    this.description = description || `${name}, an ENS name.`;
+    if (Metadata._hasNonAscii(name)) {
+      this.description +=
+        ' ⚠️ ATTENTION: This name contains non-ASCII characters as shown above. \
+Please be aware that there are characters that look identical or very \
+similar to English letters, especially characters from Cyrillic and Greek. \
+Also, traditional Chinese characters can look identical or very similar to \
+simplified variants. For more information: \
+https://en.wikipedia.org/wiki/IDN_homograph_attack';
+    }
+    this.attributes = [
+      {
+        trait_type: 'Created Date',
+        display_type: 'date',
+        value: created_date * 1000,
+      },
+    ];
+    this.name_length = name.length;
+    this.short_name = null;
+    this.length = 0;
```

Comment:
What's the difference between `name_length` and `length`?

Reply:
It's also unclear to me. that's why I left the `length` variable as a constant `0`. I should add a todo note for these not implemented variables like this one and one below in your comment.